rust/feature-flags/src/handler/session_recording.rs

```diff
@@ -0,0 +1,174 @@
+use crate::{
+    api::types::{SessionRecordingConfig, SessionRecordingField},
+    config::{Config, TeamIdCollection},
+    team::team_models::Team,
+};
+use axum::http::HeaderMap;
+use regex;
+use serde_json::{json, Value};
+
+const AUTHORIZED_MOBILE_CLIENTS: &[&str] = &[
+    "posthog-android",
+    "posthog-ios",
+    "posthog-react-native",
+    "posthog-flutter",
+];
+
+// Hard-coded values set for the session recording beta, still haven't come up with more sensible
+// defaults so these are what they are for now.
+const SAMPLE_RATE_FULL: &str = "1.00";
+const CANVAS_FPS_DEFAULT: u8 = 3;
+const CANVAS_QUALITY_DEFAULT: &str = "0.4";
+
+pub fn session_recording_config_response(
+    team: &Team,
+    headers: &HeaderMap,
+    config: &Config,
+) -> Option<SessionRecordingField> {
+    if !team.session_recording_opt_in || session_recording_domain_not_allowed(team, headers) {
+        return Some(SessionRecordingField::Disabled(false));
+    }
+
+    let capture_console_logs = team.capture_console_log_opt_in.unwrap_or(false);
+    let sample_rate = team.session_recording_sample_rate.as_ref().and_then(|sr| {
+        let sr_str = sr.to_string();
+        if sr_str == SAMPLE_RATE_FULL {
+            None
+        } else {
+            Some(sr_str)
+        }
+    });
+    let minimum_duration = team.session_recording_minimum_duration_milliseconds;
+
+    // linked_flag logic
+    let linked_flag = match &team.session_recording_linked_flag {
+        Some(cfg) => {
+            let key = cfg.get("key");
+            let variant = cfg.get("variant");
+            match (key, variant) {
+                (Some(k), Some(v)) => Some(json!({"flag": k, "variant": v})),
+                (Some(k), None) => Some(k.clone()),
+                _ => None,
+            }
+        }
+        None => None,
+    };
```

Comment:
@dmarticus this part introduces a breaking change
see here https://github.com/PostHog/posthog/pull/33655#discussion_r2144674448